cloud.foundry.cli/src/test/java/cloud/foundry/cli/crosscutting/util/FileUtilsTest.java

```diff
@@ -0,0 +1,539 @@
+
+        //Act
+        String actualYaml = FileUtils.readLocalFile(file.getPath());
+
+        //Verify
+        assertThat(actualYaml, is(expectedYaml));
+    }
+
+    @Test
+    public void testReadLocalFileOnDirectoryWithCommaSucceeds(@TempDir Path tempDir)
+            throws IOException {
+        //Arrange
+        String expectedYaml =   "- first" + System.lineSeparator() +
+                "- second" + System.lineSeparator() +
+                "- third" + System.lineSeparator();
+        File file = createTempFile(tempDir, "comma.path", "Extension.YMl", expectedYaml);
+
+        //Act
+        String actualYaml = FileUtils.readLocalFile(file.getPath());
+
+        //Verify
+        assertThat(actualYaml, is(expectedYaml));
+    }
+
+    @Test
+    public void testReadLocalFileOnValidFileTypesSucceeds(@TempDir Path tempDir)
+            throws IOException {
+        //Arrange
+        String expectedYaml =   "- first" + System.lineSeparator()  +
+                "- second" + System.lineSeparator()  +
+                "- third" + System.lineSeparator() ;
+        File fileYaml = createTempFile(tempDir, "","Extension.YMl", expectedYaml);
+        File fileYml = createTempFile(tempDir, "","Extension.YAML", expectedYaml);
+
+        //Act
+        String actualYaml = FileUtils.readLocalFile(fileYaml.getPath());
+        String actualYml = FileUtils.readLocalFile(fileYml.getPath());
+
+        //Verify
+        assertThat(actualYaml, is(expectedYaml));
+        assertThat(actualYml, is(expectedYaml));
+    }
+
+    @Test
+    public void testReadLocalFileOnCaseInsensitiveFileExtensionSucceeds(@TempDir Path tempDir)
+            throws IOException {
+        //Arrange
+        File file = createTempFile(tempDir, "", "Empty.YMl", "");
+
+        //Act
+        String actualYaml = FileUtils.readLocalFile(file.getPath());
+
+        //Verify
+        assertThat(actualYaml, is(""));
+    }
+
+
+    //TODO: disabling this test for now,
+    // since setting readable to false might not work in some environments due to access rights
+
+    /*@Test
+    public void testReadLocalFileOnMissingFilePermissionThrowsException(@TempDir Path tempDir)
+            throws IOException {
+        //Arrange
+        File file = createTempFile(tempDir, "","Empty.yml", "");
+        file.setReadable(false, false);
+
+        //Act and Verify
+        FileNotFoundException thrown = assertThrows(FileNotFoundException.class,
+                () -> FileUtils.readLocalFile(file.getPath()));
+        assertThat(thrown.getMessage(), containsString("Permission denied"));
+    }*/
+
+    @Test
+    public void testReadLocalFileOnMissingFileThrowsException() {
+
+        //Act and Verify
+        IOException exception = assertThrows(FileNotFoundException.class,
+                () -> FileUtils.readLocalFile(RESOURCE_PATH + "Missing.yaml"));
+        assertThat(exception.getMessage(), containsString("No such file or directory"));
```

Comment:
Asserting that a FileNotFoundException has to be thrown should be sufficient. The concrete error message might differ on different plattforms.